content/docs/1.5.1/volumes-and-nodes/internal-data-structures.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Internal Data Structures
+weight: 3
+---
+
+**Table of Contents**
+- [Node Disk](#node-disk)
+  - [Filesystem-Type Disk](#filesystem-type-disk)
+  - [Block-Type Disk](#block-type-disk)
+- [Replica Directory on Node Disk](#replica-directory-on-node-disk)
+
+## Node Disk
+
+### Filesystem-Type Disk
+
+A filesystem-type disk contains a configuration file named `longhorn-disk.cfg` and a directory called `replicas` that stores replica directories for volumes. The `longhorn-disk.cfg` file contains a JSON object with a single key `diskUUID` whose value is a 36-character UUID string, as shown below:
+```
+{
+  "diskUUID": "${UUID}"
+}
+```
+
+### Block-Type Disk
+
+A block-type disk is managed by the Storage Performance Development Kit (SPDK). SPDK constructs a logical volume store (lvstore) that utilizes the super blob feature of the blobstore to hold the UUID and other metadata. The blobstore types are implemented within the blobstore itself and are persisted on disk.
+
+## Replica Directory on Node Disk
+
+When creating a Longhorn V1 volume, the Longhorn system creates its downstream replicas on filesystem-type disks. The path of a replica directory follows the pattern ${DISK_PATH}/replicas/${VOLUME_NAME}-${8-CHAR_ID}. A newly created volume replica directory will contain the following files:
+
+```
+revision.counter
+volume-head-000.img
+volume-head-000.img.meta
+volume.meta
+```
+
+- revision.counter: Records the latest revision count of the replica for the volume.
+- volume-head-000.img: The actual data file containing the volume data.
+- volume-head-000.img.meta: Stores metadata related to the volume-head-000.img file, including the file name, parent file, creation timestamp, and other attributes.
+- volume.meta: Records the attributes of the volume, such as size, current head file, parent snapshot, and other metadata.
+
+After creating multiple snapshots, the disk chain will look like this:
+
+```
+snap 1 -----> snap 2 -----> ... -----> snap N -----> volume head
```

Comment:
Should we reverse the arrow? As volume head should be the latest, so it's parent is `snap N`.

Reply:
I have gone back and forth on this in the past.

- The current arrows help us understand the chronological order of the snapshots. First, `snap1` was written, then `snap2`, now `volume-head`.
- The suggested arrows help us understand the implementation detail. `volume-head` contains a pointer to `snapn`, which contains a pointer to `snapn-1`.

My soft preference is for the suggested arrows.

```suggestion
snap 1 <----- snap 2 <----- ... <----- snap N <----- volume head
```